tests/Functional/Query/QueryBuilderTest.php

```diff
@@ -555,24 +556,21 @@ public function testPlatformDoesNotSupportCTE(): void
      */
     private function prepareExpectedRows(array $rows): array
     {
-        if (! TestUtil::isDriverOneOf('ibm_db2', 'pdo_oci', 'pdo_sqlsrv', 'oci8')) {
-            return $rows;
-        }
-
-        if (! TestUtil::isDriverOneOf('ibm_db2')) {
+        if (
+            TestUtil::isDriverOneOf('pdo_oci', 'pdo_sqlsrv', 'oci8')
+            || (TestUtil::getConnectionParams()['driverOptions'][PDO::ATTR_STRINGIFY_FETCHES] ?? false) === true
```

Comment:
I intentionally did not check for `pdo_mysql`driver as this would allow us to easily add the same job for other DB engines, if the `ATTR_STRINGIFY_FETCHES` option is relevant as well